### DISCUSSION TITLE
@RabbitListener with multiple queues and its concurrency

### DISCUSSION THREAD

**Discussion by ddinde-mdsol** [rabbitmq/rabbitmq-server#9900] [Questions] [answered]
Hello rabbitmq community

If I have multiple queues listening on the same @rabbitlistener with concurrency defined as 3-5. Will each queue have an individual set of 3-5 threads or all queues will share a total of 3-5 threads?

e.g. @RabbitListener(queues= {"Q1", "Q2", "Q3"}, concurrency="3-5")
Will Q1, Q2, and Q3 have 3-5 consumers allotted each or all queues will have only 3-5 consumers in total?

**Comment by michaelklishin** [comment, marked as answer]
@ddinde-mdsol this question is entirely Spring AMQP specific.

Queues do not "share" threads, consumer implementations do. Please ask it in the appropriate Spring AMQP forum. For example, they have [GitHub Discussions enabled](https://github.com/spring-projects/spring-amqp/discussions).
